content/posts/kubernetes-chargeback.md

```diff
@@ -0,0 +1,190 @@
+
+Using namespaces to link workloads to your cost management constructs has several more advantages.
+
+- All relevant Kubernetes objects belong to a namespace.
+- Kubernetes objects belong to **only one** namespace.
+- Namespaces are not ephemeral — They usually live months or years.
+- You'll have much fewer namespaces than pods or other resource consuming objects.
+- The namespace of an object is availeble in almost all contexts, simplifying aggregation.
+- You can set quotas on namespaces.
+
+Defining a process for creating namespaces, where attribution of the resources can be ensured is recommended, to enable automatic attribution in the future.
+
+> #### Using Annotations
+>
+> Another approach to attribute workloads to your organizational units, relies on Kubernetes [annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) which can be attached to all Kubernetes objects. This allows more flexibility, but requires more investment into aggregating data.
+>
+> You can also rely on namespaces and attach annotations on them to keep the relevant data together within Kubernetes. Additionally, defining reserved capacity using annotations on namespaces is a good way to manage everything in the same place.
+
+#### Shared Namespaces
+
+You may encounter namespaces that contain resources that should be attributed to different cost centers or organizational units. This is generally discouraged and where possible should be changed.
+
+However, there are good reasons for shared namespaces. A few teams may together run infrastructure supporting their development process, such as build servers or artifact management repositories. This practice allows them to synergize management of these components. Some of these components use a lot of resources regardless of load and sharing them reduces the cost on the whole organization. While there are many options to split the cost (new cost centers, defining a split using annotations on the namespace), just billing to one of the cost centers is recommended, unless these shared namespaces turn out to be a major cost driver in your cluster.
+
+In some cases, shared namespace may also highlight a rigid cost center structure or slow and complicated processes around it or the creation of namespaces, causing namespaces to be shared or even reused instead of decommissioned. Simplifying these tasks could reduce the desire to create shared namespaces or reuse them.
+
+Additionally, you'll have namespaces that contain resources necessary for Kubernetes to run or additional tooling, such as logging and monitoring infastructure. It makes sense to calculate this cost into maintenance cost of the cluster, which you either bill to a separate cost center or add as surcharge to the pricing of your resources.
```

Comment:
`Kubernetes to run or additional tooling, such as logging and monitoring infastructure.` --> `Kubernetes or additional tooling, such as logging and monitoring infastructure to run.`

Reply:
How about leaving out "to run" alltogether?
Additionally, you'll have namespaces that contain resources necessary for Kubernetes or additional tooling, such as logging and monitoring infrastructure.